src/lightning/data/cache/pytree.py

```diff
@@ -0,0 +1,567 @@
+# Taken from PyTorch https://github.com/pytorch/pytorch/blob/e9ebda29d87ce0916ab08c06ab26fd3766a870e5/torch/utils/_pytree.py
```

Comment:
don't we have pytree already in other places? Would it make sense to unify it in utilities?

Reply:
We need to latest Pytorch version as it contains the logic to serialize the pytree to jsons.

Reply:
and since PT 2.1 is released now based on your comment we should be able to remove it now?